src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/HaplotypeCaller.java

```diff
@@ -0,0 +1,244 @@
+ *     to the desired levels in the next step of the workflow (GenotypeGVCFs)</li>
+ * </ul>
+ *
+ */
+@CommandLineProgramProperties(
+        summary = "Call germline SNPs and indels via local re-assembly of haplotypes",
+        oneLineSummary = "Call germline SNPs and indels via local re-assembly of haplotypes",
+        programGroup = VariantProgramGroup.class
+)
+public class HaplotypeCaller extends ReadWindowWalker {
+
+    @ArgumentCollection
+    private HaplotypeCallerArgumentCollection hcArgs = new HaplotypeCallerArgumentCollection();
+
+    /**
+     * A raw, unfiltered, highly sensitive callset in VCF format.
+     */
+    @Argument(fullName = StandardArgumentDefinitions.OUTPUT_LONG_NAME, shortName = StandardArgumentDefinitions.OUTPUT_SHORT_NAME, doc = "File to which variants should be written")
+    public String outputVCF = null;
+
+    private VariantContextWriter vcfWriter;
+
+    private HaplotypeCallerEngine hcEngine;
+
+    @Override
+    public boolean requiresReference() { return true; }
+
+    @Override
+    protected int defaultWindowSize() { return 300; }
+
+    @Override
+    protected int defaultWindowPadding() { return 100; }
+
+    @Override
+    public CountingReadFilter makeReadFilter() {
+        return HaplotypeCallerEngine.makeStandardHCReadFilter(hcArgs, getHeaderForReads());
+    }
+
+    @Override
+    public void onTraversalStart() {
+        hcEngine = new HaplotypeCallerEngine(hcArgs, getHeaderForReads(), referenceArguments.getReferenceFileName());
+
+        // The HC engine will make the right kind (VCF or GVCF) of writer for us
+        vcfWriter = hcEngine.makeVCFWriter(outputVCF, getHeaderForReads().getSequenceDictionary());
+        hcEngine.writeHeader(vcfWriter);
+    }
+
+    @Override
+    public void apply( ReadWindow window, ReferenceContext referenceContext, FeatureContext featureContext ) {
+        for ( final AssemblyRegion assemblyRegion : readWindowToAssemblyRegions(window) ) {
+            List<VariantContext> callsInRegion = hcEngine.callRegion(assemblyRegion, featureContext);
+
+            for ( final VariantContext call : callsInRegion ) {
+                // Only include calls that start within the window (as opposed to the padded regions around it)
+                if ( window.startsWithinWindow(call) ) {
+                    vcfWriter.add(call);
+                }
+            }
+        }
+    }
+
+    /**
+     * Convert a {@link ReadWindow} into a List of {@link AssemblyRegion} for processing by the HaplotypeCaller,
+     * covering the span of the original window
+     *
+     * @param window window to convert into AssemblyRegions
+     * @return a List of {@link AssemblyRegion} for processing by the HaplotypeCaller, covering the span of
+     *         the original window
+     */
+    private List<AssemblyRegion> readWindowToAssemblyRegions( final ReadWindow window ) {
+        List<AssemblyRegion> regions = new ArrayList<>();
```

Comment:
It's only always size 1 for now -- the 1:1 window to region correspondence is likely a temporary thing.